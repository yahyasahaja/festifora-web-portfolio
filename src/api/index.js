import axios from 'axios';
// import getPostsMock from '../mocks/get-posts.json';
import { API_KEY, API_URL, BLOG_ID } from '../configs/constants';

const interfaces = {
  getPosts() {
    return axios.get(`${API_URL}/${BLOG_ID}/posts?key=${API_KEY}`);
  }
}

export default interfaces;
