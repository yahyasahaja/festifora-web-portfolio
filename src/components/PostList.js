import React from 'react';
import api from '../api';
import PostCard from './PostCard';

const PostList = () => {
  const [ posts, setPosts ] = React.useState([]);

  React.useEffect(() => {
    const fetchData = async () => {
      try {
        const { data } = await api.getPosts();
        console.log('result', data.items);
        setPosts(data.items);
      } catch (error) {
        console.error(error);
      }
    }

    fetchData();
  }, []);

  return (
    <section data-testid="post-list">
      <h2 className="[ text-3xl text-center ]">Portfolio</h2>
      <div className="[ flex justify-center flex-wrap ] p-10">
        {posts.map((post, i) => <PostCard key={i} title={post.title} content={post.content} />)}
      </div>
    </section>
  )
}

export default PostList;
