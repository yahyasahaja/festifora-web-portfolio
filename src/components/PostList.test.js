import { render, screen } from '@testing-library/react';
import flushPromises from 'flush-promises';
import getPostsMock from '../mocks/get-posts.json';
import PostList from './PostList';
import api from '../api';

test('renders the title and content correctly', async () => {
  const mock = jest.spyOn(api, 'getPosts');
  mock.mockImplementation(() => Promise.resolve(getPostsMock));
  render(<PostList />);

  await flushPromises();

  const wrapper = screen.getByTestId('post-list');
  expect(wrapper).toMatchSnapshot();
});
