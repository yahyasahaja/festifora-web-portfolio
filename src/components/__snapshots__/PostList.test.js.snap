// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the title and content correctly 1`] = `
<section
  data-testid="post-list"
>
  <h2
    class="[ text-3xl text-center ]"
  >
    Portfolio
  </h2>
  <div
    class="[ flex justify-center flex-wrap ] p-10"
  >
    <section
      class="mt-10 m-5 p-5 border border-white rounded-2xl"
    >
      <h2
        class="text-xl font-bold"
        data-testid="title"
      >
        Latest updates, August 1st
      </h2>
      <div
        class="mt-5"
        data-testid="content"
      >
        elided for readability
      </div>
    </section>
    <section
      class="mt-10 m-5 p-5 border border-white rounded-2xl"
    >
      <h2
        class="text-xl font-bold"
        data-testid="title"
      >
        Latest updates, August 1st
      </h2>
      <div
        class="mt-5"
        data-testid="content"
      >
        elided for readability
      </div>
    </section>
  </div>
</section>
`;
